cmd/kubeadm/app/util/apiclient/idempotency.go

```diff
@@ -40,20 +40,34 @@ import (
 // ConfigMapMutator is a function that mutates the given ConfigMap and optionally returns an error
 type ConfigMapMutator func(*v1.ConfigMap) error
 
+// apiCallRetryInterval holds a local copy of apiCallRetryInterval for testing purposes
+var apiCallRetryInterval = constants.KubernetesAPICallRetryInterval
+
 // TODO: We should invent a dynamic mechanism for this using the dynamic client instead of hard-coding these functions per-type
 
 // CreateOrUpdateConfigMap creates a ConfigMap if the target resource doesn't exist. If the resource exists already, this function will update the resource instead.
 func CreateOrUpdateConfigMap(client clientset.Interface, cm *v1.ConfigMap) error {
-	if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Create(context.TODO(), cm, metav1.CreateOptions{}); err != nil {
-		if !apierrors.IsAlreadyExists(err) {
-			return errors.Wrap(err, "unable to create ConfigMap")
-		}
-
-		if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Update(context.TODO(), cm, metav1.UpdateOptions{}); err != nil {
-			return errors.Wrap(err, "unable to update ConfigMap")
-		}
+	var lastError error
+	err := wait.PollUntilContextTimeout(context.Background(),
+		apiCallRetryInterval, kubeadmapi.GetActiveTimeouts().KubernetesAPICall.Duration,
+		true, func(_ context.Context) (bool, error) {
+			ctx := context.Background()
```

Comment:
this is explained in the release note too.
we don't want to use the context passed from here:
```
 func(_ context.Context) (bool, error) {
```

because it changes expectations of the caller site if they are expecting e.g. an API error, but instead would get a context timeout error as the "last error"